src/main/java/PandaBot.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Represents the main driver of the PandaBot program.
+ */
+public class PandaBot {
+    private Ui ui;
+    private TaskList tasks;
+    private Storage storage;
+
+    /**
+     * Creates a new PandaBot object.
+     * @param fileName the file name of the save file
+     */
+    public PandaBot(String fileName) {
+        ui = new Ui();
+        storage = new Storage(fileName);
+        tasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Runs the PandaBot program.
+     */
+    public void run() {
+        ui.printWelcome();
+        boolean isExit = false;
+
+        // runs the program
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCmd();
+                ui.printLine(); // show the divider line ("_______")
```

Comment:
The word "line" is ambiguous, as in system.out.println. Maybe call it printDivider instead?